src/onboarding/registration/NameAndPicture.test.tsx

```diff
@@ -245,6 +245,25 @@ describe('NameAndPictureScreen', () => {
     expect(AccountActions.setName).not.toHaveBeenCalled()
   })
 
+  it('shows generator button and generator button puts random username in', () => {
+    mockStatsigGet.mockReturnValueOnce(false).mockReturnValueOnce('autogenerator')
```

Comment:
as you pointed out, this is very implementation dependent. Why not use `mockImplementation` and make the output different depending on the input? 

e.g. if you're getting the nameType feature flag, give 'autogenerator', but if you're getting some boolean-valued feature flag, return 'false'